site/content/en/concepts/setters/_index.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "Setters"
+linkTitle: "Setters"
+weight: 4
+type: docs
+description: >
+  Setters and related concepts
+---
+
+## Setters
+
+Setters provide a solution for template-free setting or substitution of field
+values through package metadata (OpenAPI). They are a safer alternative to other
+substitution techniques which do not have the context of the structured data
+– e.g. using `sed` to replace values.
+
+The OpenAPI definitions for setters are defined in a Kptfile and referenced by a
+fields through comments on the fields.
+
+Setters may be invoked to programmatically modify the configuration using `kpt cfg set`
+to set and/or substitute values.
+
+#### Auto setters
+
+Some setters are automatically set deriving the values from the output of `gcloud config list`
+command, when the package is fetched using `kpt pkg get`. Package consumers need not
+invoke `kpt cfg set` on them explicitly. Following are the names of supported auto-setters:
+
+```
+gcloud.core.project
+gcloud.project.projectNumber
+gcloud.compute.region
+gcloud.compute.zone
+```
+
+#### Inherit Setter Values from Parent Package
+
+##### Notice: This is an experimental feature and is subjected to changes soon
+
+When a remote kpt package is fetched(/updated) into local directory, kpt searches
+for its closest parent directory(if any) with a `Kptfile` in the file system and
+auto-fills matching setters in the fetched(/updated) package with the setter values
+from the parent package. Setters are matched based on setter names.
+
+e.g. If `mysql` package is fetched into the local directory tree of `wordpress`
```

Comment:
I think we should try to visualize this example rather than just describing the directory structure for users.